internal/system/agent/utils.go

```diff
@@ -35,3 +35,12 @@ func ProcessResponse(response string) map[string]interface{} {
 	}
 	return rsp
 }
+
+func ProcessFetchedResponse(response []byte) map[string]interface{} {
```

Comment:
IMHO this function is largely unnecessary - it doesn't return an error so a caller won't know if the operation actually succeeded or if the response really was just an empty map, and in addition you don't really save many lines of code by having the error check from this function (if this function returned an error which it should) in addition to the error check inside this function.